primefaces/src/main/java/org/primefaces/model/DefaultLazyDataModel.java

```diff
@@ -68,19 +68,7 @@ public List<T> load(int first, int pageSize, Map<String, SortMeta> sortBy, Map<S
 
         sort(filteredValues);
 
-        if (pageSize == 0) {
-            return filteredValues;
-        }
```

Comment:
Removed as strange decision (if explanation given I can put it back, but that's not how lazymodel has been working so far)

Reply:
Ah must be the case, if rows = 0 component display everything, right?

Reply:
Yep, i think so